src/gtasa/effects/custom/ped/GiveNPCsAnWeaponEffect.cpp

```diff
@@ -0,0 +1,56 @@
+#include "effects/OneTimeEffect.h"
+
+#include <vector>
+
+#include <CPickups.h>
+#include <CStreaming.h>
+#include <extensions/ScriptCommands.h>
+
+using namespace plugin;
+
+using weapon_list = std::initializer_list<int>;
+
+class GiveNPCsAnWeaponEffect : public OneTimeEffect
+{
+private:
+    std::vector<int> weapons;
+
+public:
+    GiveNPCsAnWeaponEffect (weapon_list weaponsList) : weapons (weaponsList) {}
+
+    void
+    OnStart (EffectInstance *inst) override
+    {
+        CPlayerPed *player = FindPlayerPed ();
+
+        for (auto w : weapons)
+        {
+            int m = CPickups::ModelForWeapon (eWeaponType (w));
+            CStreaming::RequestModel (m, 2);
+        }
+
+        CStreaming::LoadAllRequestedModels (false);
+
+        for (CPed *ped : CPools::ms_pPedPool)
+        {
+            if (ped == player) continue;
+
+            auto w = weapons[inst->Random (1, 100000) % weapons.size ()];
+            Command<eScriptCommands::COMMAND_GIVE_WEAPON_TO_CHAR> (ped, w,
+                                                                   9999);
+            Command<eScriptCommands::COMMAND_SET_CURRENT_CHAR_WEAPON> (ped, w);
+        }
+
+        for (auto w : weapons)
+        {
+            int m = CPickups::ModelForWeapon (eWeaponType (w));
+            CStreaming::SetModelIsDeletable (m);
```

Comment:
Tested this in dev right now, you can safely move it underneath line 29 (`RequestModel`) so no need for a 2nd loop